packages/amplify-cli/src/__tests__/commands/init.test.ts

```diff
@@ -36,14 +36,11 @@ jest.mock('which');
 (existsSync as jest.Mock).mockReturnValue(true);
 (readdirSync as jest.Mock).mockReturnValue([]);
 (sync as jest.MockedFunction<typeof sync>).mockReturnValue('mock/path');
-(getPackageManager as jest.MockedFunction<typeof getPackageManager>).mockReturnValue(
-  new Promise((resolve) => {
-    const packageManager = new YarnPackageManager();
-    packageManager.lockFile = 'mock.lock';
-    packageManager.version = coerce('1.22.0') ?? undefined;
-    resolve(packageManager);
-  }),
-);
+
+const packageManager = getPackageManagerByType('yarn');
+packageManager.lockFile = 'mock.lock';
```

Comment:
Once you make `lockFile` readonly on the interface you won't be able to do this, but I think that's okay

Reply:
that is the only reason I didn't make it readonly on the interface, if we are good to introduce as $TSAny, for me to keep the tests running then I can update